### DISCUSSION TITLE
pyannote gives wrong results

### DISCUSSION THREAD

**Discussion by amitli1** [pyannote/pyannote-audio#1290] [Development]
I'm trying to use `pyannote` for `speaker diarization`
and I'm getting wrong number of speakers. 

Any example I tried I got wrong results.

For example:

- I used this youtube file:
[https://www.youtube.com/watch?v=b2_ZZ2UpSzI][1]

- I convert it to wav file with sample rate of 16000.

I run the following code:

    from pyannote.audio import Pipeline
    
    TEST_FILE = "example.wav"
    MY_TOKEN = "..."
    pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization",
                                        use_auth_token=MY_TOKEN)
    
    diarization = pipeline(TEST_FILE)

And I got the following diarization:
[![enter image description here][2]][2]

- The GT contains 4 speakers and not 2.

How can I tweak `pyannote` and get better results ?

  [1]: https://www.youtube.com/watch?v=b2_ZZ2UpSzI
  [2]: https://i.stack.imgur.com/PeUCU.png

**Comment by joagonzalez** [comment]
have you tried helping the pipeline with _num_speakers_ parameter?


```
pipeline = Pipeline.from_pretrained(
                     self.model,
                     use_auth_token=self.token,
                 )        
        
diarization = pipeline(
    filename, 
    num_speakers=self.num_speakers
)
```

**Reply by amitli1** [reply]
Now it separate to 3 different speakers, but with very bad separation:
![image](https://user-images.githubusercontent.com/128060257/227199526-f3f3005d-f869-41c7-b6ae-cf1ebf2b5a6a.png)

What are my options? (I'm using a simple wav file, so I assumed it need to be an easy task for pyannote)

**Reply by hbredin** [reply]
See FAQ related to [how to improve performance](https://github.com/pyannote/pyannote-audio/blob/develop/FAQ.md#how-can-i-improve-performance), maybe?
